### PR TITLE
[CPP20][Visualisation] Fix deprecated enum bitwise operation in FWCandidateHGCalLegoProxyBuilder

### PR DESCRIPTION
**Pull request by iarspider**
#### PR description:

Fix deprecated enum bitwise operation in FWCandidateHGCalLegoProxyBuilder

```
src/Fireworks/Candidates/plugins/FWCandidateHGCalLegoProxyBuilder.cc: In static member function 'static const std::string& FWCandidateHGCalLegoProxyBuilder::classView()':
  src/Fireworks/Candidates/plugins/FWCandidateHGCalLegoProxyBuilder.cc:97:46: warning: bitwise operation between different enumeration types 'FWViewType::EBit' and 'FWViewType::EType' is deprecated [-Wdeprecated-enum-enum-conversion]
    97 |                         FWViewType::kLegoBit | FWViewType::kLegoHF);
      |                         ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
```

#### PR validation:

Bot tests

### CODE REVIEW

**Review by alja — APPROVED** [review]

**Comment by alja** [comment]
@makortel This enum is used for different view types. E.g. 3D, projected views, glimpse , histogram, table type views etc. There are different types of histogram (or lego) views. Same as there are different types of projected views. Some of graphic scene elements are shared within similar view types.

**Comment by makortel** [comment]
How would `kLegoBit` be distinguished from `k3DBit | kRhoPhiBit`? They both have the same numerical value (5).

**Comment by alja** [comment]
@makortel 

There is bitwise left shit operators
kLegoBit = 16
k3DBit | kRhoPhiBit = 3

I agree the whole process of plugin registration and the operators in quite convoluted.

**Comment by makortel** [comment]
> There is bitwise left shit operators

Ah sorry, I meant

> How would `kLegoHF` be distinguished from `k3DBit | kRhoPhiBit`? They both have the same numerical value (5).

It is specifically this `or` of `EType` value and `EBit` value
https://github.com/cms-sw/cmssw/blob/ab77a26b8f468cc38636c60812dff1082c6fa645/Fireworks/Candidates/plugins/FWCandidateHGCalLegoProxyBuilder.cc#L97
that I'm wondering about.

**Comment by alja** [comment]
-1

**Comment by alja** [comment]
@makortel I see .. It should be  
```
FWViewType::kLegoBit | FWViewType::kLegoHFBit
); 
```

**Comment by alja** [comment]
@makortel @iarspider What should we do with this PR?

**Comment by makortel** [comment]
In my opinion it would be good to replace the `EBit` enum with `static constexpr unsigned int` constants, and then fix the line as @alja suggested in https://github.com/cms-sw/cmssw/pull/45076#issuecomment-2136163268 . It doesn't really matter to me whether those would be done in this PR or in a separate PR (in which case I'd close this PR).

**Comment by alja** [comment]
> In my opinion it would be good to replace the `EBit` enum with `static constexpr unsigned int` constants, and then fix the line as @alja suggested in [#45076 (comment)](https://github.com/cms-sw/cmssw/pull/45076#issuecomment-2136163268) . It doesn't really matter to me whether those would be done in this PR or in a separate PR (in which case I'd close this PR).

OK, I will make the change mentioned in [#45076 (comment)](https://github.com/cms-sw/cmssw/pull/45076#issuecomment-2136163268) in a separate PR.

**Comment by iarspider** [comment]
please test for CMSSW_14_1_CPP20_X

**Comment by iarspider** [comment]
please test

File: Fireworks/Core/interface/FWViewType.h [hunk 1]
```diff
@@ -46,22 +46,11 @@ class FWViewType {
...
-    kLegoHFBit = 1 << kLegoHF,
-    kLegoPFECALBit = 1 << kLegoPFECAL,
-    kGlimpseBit = 1 << kGlimpse,
-    kTableBit = 1 << kTable,
-    kTableHLTBit = 1 << kTableHLT,
-    kTableL1Bit = 1 << kTableL1,
-    kGeometryBit = 1 << kGeometryTable,
-    kOverlapBit = 1 << kOverlapTable
-  };
+  static constexpr int k3DBit = 1 << k3D, kRhoPhiBit = 1 << kRhoPhi, kRhoZBit = 1 << kRhoZ,
```
Comment (makortel) [inline Fireworks/Core/interface/FWViewType.h:49]: Please use `unsigned int` (behavior of bit operations is easiest to understand on unsigned)

File: Fireworks/Candidates/plugins/FWCandidateHGCalLegoProxyBuilder.cc [hunk 2]
```diff
@@ -94,4 +94,4 @@ void FWCandidateHGCalLegoProxyBuilder::build(const reco::HGCalMultiCluster &iDat
 REGISTER_FWPROXYBUILDER(FWCandidateHGCalLegoProxyBuilder,
                         reco::HGCalMultiCluster,
                         "HGCal Multiclusters Lego",
-                        FWViewType::kLegoBit | FWViewType::kLegoHF);
+                        static_cast<int>(FWViewType::kLegoBit) | static_cast<int>(FWViewType::kLegoHFBit));
```
Comment (makortel) [inline Fireworks/Candidates/plugins/FWCandidateHGCalLegoProxyBuilder.cc:97]: Could the `static_cast` be removed now?

@alja Note that presently this PR includes the fix mentioned in https://github.com/cms-sw/cmssw/pull/45076#issuecomment-2136163268

File: Fireworks/Core/interface/FWViewType.h (hunk 1 above)
Comment (iarspider) [reply Fireworks/Core/interface/FWViewType.h]: done

File: Fireworks/Candidates/plugins/FWCandidateHGCalLegoProxyBuilder.cc (hunk 2 above)
Comment (iarspider) [reply Fireworks/Candidates/plugins/FWCandidateHGCalLegoProxyBuilder.cc]: done

**Comment by iarspider** [comment]
please test

**Comment by iarspider** [comment]
@cms-sw/visualization-l2 gentle ping

**Comment by alja** [comment]
+1

**Comment by rappoccio** [comment]
+1

**Pull request merged**